src/breaker/CountBreaker.ts

```diff
@@ -0,0 +1,124 @@
+import { CircuitState } from '../CircuitBreakerPolicy';
+import { IBreaker } from './Breaker';
+
+export interface ICountBreakerOptions {
+  /**
+   * Percentage (from 0 to 1) of requests that need to fail before we'll
+   * open the circuit.
+   */
+  threshold: number;
+
+  /**
+   * Size of the count based sliding window.
+   */
+  size: number;
+
+  /**
+   * Minimum number of calls needed to (potentially) open the circuit.
+   * Useful to avoid unnecessarily tripping when there are only few samples yet.
+   * Defaults to {@link ICountBreakerOptions.size}.
+   */
+  minimumNumberOfCalls?: number;
+}
+
+export class CountBreaker implements IBreaker {
+  private readonly threshold: number;
+  private readonly minimumNumberOfCalls: number;
+
+  /**
+   * The samples in the sliding window. `true` means "success", `false` means
+   * "failure" and `undefined` means that there is no sample yet.
+   */
+  private readonly samples: (boolean | undefined)[];
+  private successes = 0;
+  private failures = 0;
+  private currentSample = 0;
+
+  /**
+   * CountBreaker breaks if more than `threshold` percentage of the last `size`
+   * calls failed, so long as at least `minimumNumberOfCalls` calls have been
+   * performed (to avoid opening unnecessarily if there are only few samples
+   * in the sliding window yet).
+   */
+  constructor({ threshold, size, minimumNumberOfCalls = size }: ICountBreakerOptions) {
+    if (threshold <= 0 || threshold >= 1) {
+      throw new RangeError(`CountBreaker threshold should be between (0, 1), got ${threshold}`);
+    }
+    if (!Number.isSafeInteger(size) || size < 1) {
+      throw new RangeError(`CountBreaker size should be a positive integer, got ${size}`);
+    }
+    if (
+      !Number.isSafeInteger(minimumNumberOfCalls) ||
+      minimumNumberOfCalls < 1 ||
+      minimumNumberOfCalls > size
+    ) {
+      throw new RangeError(
+        `CountBreaker size should be an integer between (1, size), got ${minimumNumberOfCalls}`,
+      );
+    }
+
+    this.threshold = threshold;
+    this.minimumNumberOfCalls = minimumNumberOfCalls;
+    this.samples = Array.from<undefined>({ length: size }).fill(undefined);
+  }
+
+  /**
+   * @inheritdoc
+   */
+  public success(state: CircuitState) {
+    if (state === CircuitState.HalfOpen) {
+      this.reset();
+    }
+
+    this.sample(true);
+  }
+
+  /**
+   * @inheritdoc
+   */
+  public failure(state: CircuitState) {
+    this.sample(false);
+
+    if (state !== CircuitState.Closed) {
+      return true;
+    }
+
+    const total = this.successes + this.failures;
+
+    if (total < this.minimumNumberOfCalls) {
+      return false;
+    }
+
+    if (this.failures > this.threshold * total) {
```

Comment:
It's wobbly because of floating point stuff, but

```suggestion
    if (this.failures >= this.threshold * total) {
```

is more correct

Reply:
`this.failures` is always an integer, so floating point stuff shouldn't be relevant, I think. But I'm also fine with including the `=`, just need to adjust the wording a bit then ("when threshold is reached" instead of "when threshold is surpassed" etc.).

I mostly just did it this way because [it's also done like that in the SamplingBreaker](https://github.com/connor4312/cockatiel/blob/master/src/breaker/SamplingBreaker.ts#L104). Should I change it there, too?

Reply:
Ah, okay, matching that behavior is good then 👍